pkg/cmd/cli/cmd/login.go

```diff
@@ -178,6 +183,43 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 	return nil
 }
 
+func (o LoginOptions) checkConfigIsWritable(f *osclientcmd.Factory) (bool, error) {
+	kubeconfig := o.PathOptions.GetDefaultFilename()
+
+	file, err := os.OpenFile(kubeconfig, os.O_CREATE|os.O_WRONLY, 0600)
+	if err == nil {
+		file.Close()
+		return true, nil
+	}
+
+	// Whole directory does not exist and we cannot create kubeconfig file.
+	if os.IsNotExist(err) {
+		return true, nil
+	}
+
+	errmsg := kubeconfig
+
+	if os.IsPermission(err) {
+		errmsg += ": The kubeconfig file is not writable. Specify a writeable path with --config or by setting $KUBECONFIG."
```

Comment:
> @legionus Do you prefer this flow with CheckErr or the one with simply writing to stderr?

If you're asking my opinion, I think almost  this whole idea If you're asking my opinion, I think almost this whole idea wrong.

1. This pre-check are not equivalent to creating a config file. This pre-check does not check the directories. [WriteToFile](https://github.com/openshift/origin/blob/388478c40e751c4295dcb9a44dd69e5ac65d0e3b/Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/loader.go#L288-L303) uses os.MkdirAll to create parent directories. So if some of the directories will not be available for recording, we will get EPERM.

1. With this pre-check we have TOCTOU. The problem may appear between the check of write access and trying to write.

1. Due to the fact that we can't reproduce the creation of the file we can't stop on the first error from `checkConfigIsWritable`.

1. Getting current username doesn't always work because for this you need a configuration file, but we haven't created in some cases. In this case, we have `system:anonymous` as user and our warning for `system:admin` will not be displayed.

I think correct would be to add additional `IsPermission` here: https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/login.go#L81-L83 and display a message which suggested by Jordan.